plugins/rikki/heroeslounge/components/SlothAccount.php

```diff
@@ -279,26 +279,27 @@ public function onRegister()
 
     public function onParticipationSave()
     {
-        try {
-            foreach ($this->seasons as $season) {
-                if ($season->current_round == 0) {
-                    $val = post('part_seas_'.$season->id);
-                    if (isset($val) && !$season->free_agents->contains($this->sloth->id)) {
-                        $season->free_agents()->attach($this->sloth->id);
-                        Discord\RoleManagement::UpdateUserRole("PUT", $this->sloth->discord_id, "FreeAgent");
-                        Flash::success('You will participate in '.$season->title.'!');
-                    } elseif (isset($val) == false) {
-                        $season->free_agents()->detach($this->sloth->id);
-                        Discord\RoleManagement::UpdateUserRole("DELETE", $this->sloth->discord_id, "FreeAgent");
-                        Flash::error('You will not paricipate in '.$season->title.' - Sad to see you go!');
-                    }
+        foreach ($this->seasons as $season) {
+            if ($season->reg_open == 1) {
+                if ($this->sloth->isInTeamParticipatingInSeason($season)) {
+                    Flash::error('You can not update your '.$season->title.' participation, because you are already participating with a team!');
+                    return Redirect::refresh(); 
+                }
+
+                $participation = post('part_seas_'.$season->id);
+                if (isset($participation) && !$season->free_agents->contains($this->sloth->id)) {
+                    $season->free_agents()->attach($this->sloth->id);
+                    Discord\RoleManagement::UpdateUserRole("PUT", $this->sloth->discord_id, "FreeAgent");
+                    Flash::success('You will participate in '.$season->title.'!');
+                } elseif (isset($participation) == false) {
+                    $season->free_agents()->detach($this->sloth->id);
+                    Discord\RoleManagement::UpdateUserRole("DELETE", $this->sloth->discord_id, "FreeAgent");
+                    Flash::error('You will not participate in '.$season->title.' - Sad to see you go!');
                 }
             }
-        } catch (Exception $e) {
-            Flash::error($e->getMessage());
-        } finally {
-            return Redirect::refresh();
         }
+
+        return Redirect::refresh();
```

Comment:
General comment: We can just get rid of the entire "Sign up on your profile" concept

Reply:
So would we still want users to be able to remove their participation from a season on their profile? Or would we want this to also be removed and added somewhere else?

Reply:
Ideally, all of the signing up / remove participation should be on the season / playoff page

Reply:
Updated the free agent participation changes to be on season pages. I did keep a listing of participation on the profile page for legacy reasons.

I didn't change anything for playoffs, as we don't allow free agent sign ups there anyway.